src/test/util_threadnames_tests.cpp

```diff
@@ -52,11 +50,6 @@ std::set<std::string> RenameEnMasse(int num_threads)
  */
 BOOST_AUTO_TEST_CASE(util_threadnames_test_rename_threaded)
 {
-#if !defined(HAVE_THREAD_LOCAL)
-    // This test doesn't apply to platforms where we don't have thread_local.
-    return;
-#endif
-
     std::set<std::string> names = RenameEnMasse(100);
```

Comment:
With this it will be possible to remove `export NOWARN_CXXFLAGS="-Wno-error=unreachable-code"` from dev environment on FreeBSD, thanks!